controlplane/kubeadm/internal/kubeadm_config_map_test.go

```diff
@@ -86,3 +89,97 @@ kind: ClusterStatus`,
 		)),
 	))
 }
+
+func Test_kubeadmConfig_UpdateCoreDNSImageInfo(t *testing.T) {
+	cm := &corev1.ConfigMap{
+		Data: map[string]string{
+			"ClusterConfiguration": `apiServer:
+  extraArgs:
+    authorization-mode: Node,RBAC
+    cloud-provider: aws
+  timeoutForControlPlane: 4m0s
+apiVersion: kubeadm.k8s.io/v1beta2
+certificatesDir: /etc/kubernetes/pki
+clusterName: foobar
+controlPlaneEndpoint: foobar.us-east-2.elb.amazonaws.com
+controllerManager:
+  extraArgs:
+    cloud-provider: aws
+dns:
+  type: CoreDNS
+etcd:
+  local:
+    dataDir: /var/lib/etcd
+imageRepository: k8s.gcr.io
+kind: ClusterConfiguration
+kubernetesVersion: v1.16.1
+networking:
+  dnsDomain: cluster.local
+  podSubnet: 192.168.0.0/16
+  serviceSubnet: 10.96.0.0/12
+scheduler: {}`,
+		},
+	}
+
+	badcm := &corev1.ConfigMap{
+		Data: map[string]string{
+			"ClusterConfiguration": `apiServer:
+  extraArgs:
+    authorization-mode: Node,RBAC
+	...`,
+		},
+	}
+
+	tests := []struct {
+		name      string
+		cm        *corev1.ConfigMap
+		expectErr bool
+	}{
+		{
+			name:      "sets the image repository and tag",
+			cm:        cm,
+			expectErr: false,
+		},
+		{
+			name:      "returns error if unable to convert yaml",
+			cm:        badcm,
+			expectErr: true,
+		},
+		{
+			name:      "returns error if cannot find cluster config key",
+			cm:        &corev1.ConfigMap{Data: map[string]string{}},
+			expectErr: true,
+		},
+	}
```

Comment:
Could we add a test for "does not set imageRepository or imageTag in the config map if they aren't specified in the KubeadmConfig"?

Reply:
Do you mean that if the imageRepo and imageTag are not defined in the ClusterConfig, then don't set them?

Reply:
Yeah – I assume that's enough to signal to kubeadm not to muck with the CoreDNS deployment (which is what I'm really after).